todo_app/app.py

```diff
@@ -26,20 +37,35 @@ def create_app():
     todos = mongo.MyDatabase[collection]
     _DEFAULT_ITEMS = []
     itemDict = []
+    if os.getenv("disable_login")=='True':
+        app.config["LOGIN_DISABLED"]=True
+
+    def check_role(func):
+        def wrapper_function(*args, **kwargs):
+            if 'writer' in current_user.roles:
+                return func(*args, **kwargs)
```

Comment:
This place would be a good place to throw an exception, e.g.
```suggestion
            if 'writer' in current_user.roles:
                return func(*args, **kwargs)
            else:
                raise Forbidden("Writer role required")
```
(`Forbidden` is from `werkzeug.exceptions`)